### PR TITLE
feat(APIM-538): log format environmental variable

### PR DESCRIPTION
**Pull request by AlexBramhill**
# Introduction
This PR adds tests around the `app.config`

# Resolution
This adds tests via our existing test helpers, as well as adding a `getBooleanConfig` function for config files

# Misc
There is no comparable helpers in `mdm` or `tfs`, as a result, there will be no PR for these repos

### CODE REVIEW

File: src/config/app.config.test.ts [hunk 1]
```diff
@@ -79,124 +77,106 @@ describe('appConfig', () => {
...
-        REDACT_LOGS: 'false',
-      });
-
-      const config = appConfig();
-
-      expect(config.redactLogs).toBe(false);
-    });
-
-    it('sets redactLogs to true if REDACT_LOGS is not specified', () => {
+  describe('parsing HTTP_VERSIONING_ENABLE', () => {
```
Comment (AlexBramhill) [inline src/config/app.config.test.ts:80]: We have this as separate tests as it is a nested object and time constraints means we are not able to rework the existing helper implementation

File: test/common-tests/environment-variable-parsing-unit-tests.ts [hunk 2]
```diff
@@ -168,45 +185,13 @@ export const withEnvironmentVariableParsingUnitTests = <ConfigUnderTest>({
...
-
-        it(`is the default value if ${environmentVariableName} is empty string`, () => {
-          process.env = {
-            [environmentVariableName]: '',
-          };
-
-          const { [configPropertyName]: configPropertyValue } = getConfig();
-
-          expect(configPropertyValue).toBe(defaultConfigValue);
-        });
```
Comment (lmarrai-sw) [inline test/common-tests/environment-variable-parsing-unit-tests.ts:209]: Why did we get rid of this test?

File: src/config/app.config.ts [hunk 3]
```diff
@@ -42,7 +44,7 @@ export default registerAs('app', (): Record<string, any> => {
     port: getIntConfig(process.env.HTTP_PORT, 3001),
     apiKey: process.env.API_KEY,
     logLevel: process.env.LOG_LEVEL || 'info',
-    redactLogs: process.env.REDACT_LOGS !== 'false',
-    singleLineLogFormat: process.env.SINGLE_LINE_LOG_FORMAT !== 'false',
+    redactLogs: getBooleanConfig(process.env.REDACT_LOGS, true),
+    singleLineLogFormat: getBooleanConfig(process.env.SINGLE_LINE_LOG_FORMAT, false),
```
Comment (lmarrai-sw) [inline src/config/app.config.ts:48]: Why did we change the default for `singleLineLogFormat`?

**Review by lmarrai-sw — DISMISSED** [review]
thanks @AlexBramhill, just a couple small things for me

File: src/config/app.config.ts (hunk 3 above)
Comment (AlexBramhill) [reply src/config/app.config.ts:48]: Ah -- Changing back

File: test/common-tests/environment-variable-parsing-unit-tests.ts (hunk 2 above)
Comment (AlexBramhill) [reply test/common-tests/environment-variable-parsing-unit-tests.ts:209]: We're now forcing defaults -- An empty string will throw a invalid boolean error (I've added a test case for this in get-boolean-config.helper.test`

**Review by lmarrai-sw — APPROVED** [review]
LGTM

**Pull request merged**